### PR TITLE
shibboleth-idp 3.3.2 (new formula)

### PR DESCRIPTION
**Pull request by romank8k**
New formula for the Shibboleth Identity Provider (IDP).

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/shibboleth-idp.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+  def install
+    # Remove Windows scripts
+    rm_rf Dir["bin/*.bat"]
+
+    # Install files
+    libexec.install Dir["*"]
+  end
+
+  test do
```
Comment (woodruffw) [inline Formula/shibboleth-idp.rb]: We require new formulae in homebrew-core to have a more substantive test than just `version` or `help`.

File: Formula/shibboleth-idp.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  homepage "https://www.shibboleth.net/"
+  url "https://shibboleth.net/downloads/identity-provider/3.3.2/shibboleth-identity-provider-3.3.2.tar.gz"
+  sha256 "ed9fbefd273199d2841d4045b2661671c53825ed3c7d52d38bfe516b39d5fc64"
+
+  bottle :unneeded
+
+  depends_on :java => "1.8+"
+
+  def install
+    # Remove Windows scripts
```
Comment (woodruffw) [inline Formula/shibboleth-idp.rb]: You can delete this comment and the comment below; the lines are pretty self-explanatory.

File: Formula/shibboleth-idp.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    rm_rf Dir["bin/*.bat"]
+
+    # Install files
+    libexec.install Dir["*"]
```
Comment (woodruffw) [inline Formula/shibboleth-idp.rb]: I'm not that familiar with shibboleth, but does it expect everything to be installed in `libexec`? Are there executables that should also be installed/linked to `bin`, for example?

**Review by woodruffw — CHANGES_REQUESTED** [review]

**Comment by woodruffw** [comment]
Thanks for the formula @rkhmelichek! I've left some review notes above.

**Comment by rkhmelichek** [comment]
Thanks for reviewing @woodruffw - I removed the comments.

The Shibboleth IDP package is meant to be used to build a war file that can then be deployed into a Tomcat or Jetty server.

Typical usage would be something like this:
```bash
export JAVA_HOME=$(/usr/libexec/java_home)
/usr/local/opt/shibboleth-idp/libexec/bin/build.sh -Didp.target.dir=/usr/local/opt/shibboleth-idp/libexec
```

And the resulting war file will be here:
`/usr/local/opt/shibboleth-idp/libexec/war/idp.war`

I tried to make that into a test, but unfortunately, the ant script that builds the war writes a log file into the Cellar `/usr/local/Cellar/shibboleth-idp/3.3.2/libexec/idp_ant.log`. And the brew test harness complains that the Cellar is read only. It wasn't possible to change the location of the log file or stop the logging without changing the ant script.

Let me know if you have suggestions on how to make this work.

File: Formula/shibboleth-idp.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  depends_on :java => "1.8+"
+
+  def install
+    rm_rf Dir["bin/*.bat"]
+
+    libexec.install Dir["*"]
+  end
+
+  test do
+    ENV["JAVA_HOME"] = `/usr/libexec/java_home`.chomp
```
Comment (DomT4) [inline Formula/shibboleth-idp.rb]: You should be writing a wrapper script so this isn't necessary. Search `write_jar_script` for examples.

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by ilovezfs** [comment]
What's the status here @rkhmelichek?

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by ilovezfs** [comment]
Please feel free to reopen when the feedback has been addressed. Thanks for the PR @rkhmelichek!

**Pull request closed without merging**